pyingest/parsers/oup.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import json
+import codecs
+import string
+from adsputils import u2asc
+from jats import JATSParser
+from pyingest.config.config import *
+from pyingest.parsers.entity_convert import EntityConverter
+from author_names import AuthorNames
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+
+class OUPJATSParser(JATSParser):
+
+    AST_WORDS = [x.lower() for x in UAT_ASTRO_KEYWORDS]
+    AST_WORDS = AST_WORDS + [x.lower() for x in AAS_ASTRO_KEYWORDS]
+    AST_WORDS = AST_WORDS + [x.lower() for x in APS_ASTRO_KEYWORDS]
+
+
+    def get_author_init(self,namestring):
+        output = u2asc(namestring)
+        for c in output:
+            if c.isalpha():
+                return c.upper()
+        return u'.'
+
+    def oup_journals(self, pid):
+        try:
+            bibstem = OUP_PUBLISHER_IDS[pid]
+        except KeyError:
+            return 'XSTEM'
+        else:
+            return bibstem
+
+    def get_tmp_page(self, bs):
+        if(bs.lower() == 'mnrasl'):
+            f = "/proj/ads/abstracts/config/links//DOI/MNRASL"
+        elif(bs.lower() == 'mnras'):
+            f = "/proj/ads/abstracts/config/links//DOI/MNRAS"
+        elif(bs.lower() == 'gji'):
+            f = "/proj/ads/abstracts/config/links//DOI/GeoJI"
+        f = f + "early.dat.nocheck"
+        with open(f,'rU') as fp:
+            p = fp.readline()
+        return p.split()[0]
+
+
+    def update_tmp_file(self, bs, bib, doi):
+        if(bs.lower() == 'mnrasl'):
```

Comment:
Same comment as above -- you wouldn't need to retype this block of code if you were getting the value from (for example) config.OUP_LINKS_DIRS